osu.Game.Rulesets.Osu/Skinning/HitPositionMeter.cs

```diff
@@ -0,0 +1,255 @@
+            });
+
+            averagePositionContainer.MoveTo(averagePosition = (relativeHitPosition + averagePosition) / 2, 800, Easing.OutQuint);
+        }
+
+        private Color4 getColourForPosition(Vector2 position)
+        {
+            switch (Vector2.Distance(position, Vector2.Zero))
+            {
+                case >= 0.5f:
+                    return colours.Red;
+
+                case >= 0.35f:
+                    return colours.Yellow;
+
+                case >= 0.2f:
+                    return colours.Green;
+
+                default:
+                    return colours.Blue;
+            }
+        }
+
+        public override void Clear()
+        {
+            averagePositionContainer.MoveTo(averagePosition = Vector2.Zero, 800, Easing.OutQuint);
+            hitPositionsContainer.Clear();
+        }
+
+        private partial class HitPosition : PoolableDrawable
+        {
+            [Resolved]
+            private HitPositionMeter hitPositionMeter { get; set; } = null!;
+
+            public readonly BindableNumber<float> JudgmentSize = new BindableFloat();
+
+            public readonly Bindable<HitPositionStyle> JudgmentStyle = new Bindable<HitPositionStyle>();
+
+            public HitPosition()
+            {
+                RelativePositionAxes = Axes.Both;
+
+                Anchor = Anchor.Centre;
+                Origin = Anchor.Centre;
+
+                InternalChildren = new Drawable[]
+                {
+                    new Circle
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Width = 0.25f,
+                        Rotation = -45
+                    },
+                    new Circle
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Width = 0.25f,
+                        Rotation = 45
+                    }
+                };
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                JudgmentSize.BindTo(hitPositionMeter.JudgmentSize);
+                JudgmentSize.BindValueChanged(size => Size = new Vector2(size.NewValue), true);
+                JudgmentStyle.BindTo(hitPositionMeter.JudgmentStyle);
+                JudgmentStyle.BindValueChanged(style => Rotation = style.NewValue == HitPositionStyle.X ? 0 : 45);
+            }
+
+            protected override void PrepareForUse()
+            {
+                base.PrepareForUse();
+
+                const int judgement_fade_in_duration = 100;
+                const int judgement_fade_out_duration = 5000;
+
+                Alpha = 0;
+
+                this
+                    .FadeTo(1f, judgement_fade_in_duration, Easing.OutQuint)
```

Comment:
`FadeInFromZero` is safer here, else you might somehow get a drawable returned to the pool which wasn't at zero alpha causing a sudden transition.

Reply:
I am already apply it, but I think it meant [BarHitErrorMeter ](https://github.com/ppy/osu/blob/master/osu.Game/Screens/Play/HUD/HitErrorMeters/BarHitErrorMeter.cs#L479)also need change?